src/composables/usePool.ts

```diff
@@ -344,30 +371,74 @@ export function flatTokenTree(
       ) {
         tokens.push(token.token.pool.tokens[token.token.pool.mainIndex]);
       } else {
-        const nestedTokens = flatTokenTree(token.token.pool?.tokens, options);
+        const nestedTokens = flatTokenTree(token.token.pool, options);
         tokens.push(...nestedTokens);
       }
     }
+  });
+
+  // Avoid duplicated tokens with the same address
+  return uniqWith(tokens, (token1, token2) =>
+    isSameAddress(token1.address, token2.address)
+  );
+}
+
+export function removePremintedBPT(pool: Pool | TokenTreePool) {
+  let newPool: Pool | TokenTreePool;
+  if (isTokenTreePool(pool)) {
+    //Avoid cloning when TokenTreePool for performance reasons
+    newPool = pool;
+  } else {
+    newPool = cloneDeep(pool);
+  }
+
+  if (newPool.tokens) {
+    removePremintedToken(newPool);
+
+    newPool.tokens.forEach(token => {
+      if (token.token.pool) {
+        removePremintedBPT(token.token.pool) as TokenTreePool;
+      }
+    });
+  }
+  return newPool;
+}
+
+function removePremintedToken(pool: Pool | TokenTreePool) {
```

Comment:
It does not return because it is a mutating function that updated the passed TokenTreePool by reference (like, for example, splice does). I agree that it is better to have pure inmutable functions that return a new object but, in this case, it was done for performance reasons (and it is a private function and it is tested so it should be ok).